platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/ResultRepository.java

```diff
@@ -28,4 +28,12 @@
             "JOIN challenge.challengeTemplate template " +
             "WHERE template.id = :templateId ")
     List<Result> findAllByTemplate(@Param("template") UUID templateId);
+
+    @Query(" SELECT result, count(taskResult), max(taskResult.endTime) FROM Result result " +
+            "JOIN result.challenge challenge " +
+            "JOIN result.taskResults taskResult " +
+            "WHERE challenge.id =:challengeId and taskResult.green=true " +
+            "GROUP BY result.user " +
+            "ORDER BY COUNT(taskResult) DESC, MAX(taskResult.endTime) ASC ")
```

Comment:
you count all taskResults? `and taskResult.green=true ` this does not mean you select only green taskResults..it means you select results with green taskResults